CONTRIBUTING.md

```diff
@@ -2,22 +2,26 @@
+If you're interested in contributing to this repo, please work through these steps carefully.  Bad pull requests waste a lot of time, and these steps will ensure your PR is a good one!  This guide is written with beginners in mind, but we would appreciate if experienced developers work through this at least once so we're all on the same page!
 
 Save us all some time and frustration by working through these steps carefully, at least once, and understand them all!
 
 ### Running Tests and Checking Code Style
-You can run tests either locally, or through the web container.:
-1. This will run all the project's tests and if successful, will also check the code style using flake 8 (make sure you're in your virtual environment):
-   * using venv: `./src/manage.py test src && flake8 src`
-   * using docker: `docker-compose exec web bash -c "./src/manage.py test src && flake8 src"`  (assuming it's running. If not, change `exec` to `run`)
-2. Tests take too long, but you can speed them up a number of ways:
+You can run tests either locally, or through the web container:
+1. This will run all the project's tests and if successful, will also check the code style using flake8 (make sure you're in your virtual environment):
+   * using venv: `python src/manage.py test src && flake8 src`
+   * using docker: `docker-compose exec web bash -c "python src/manage.py test src && flake8 src"`  (assuming it's running. If not, change `exec` to `run`)
+1. Tests take too long, but you can speed them up a number of ways:
    * Quit after the first error or failure, and also by running th tests in parallel to take advantage of multi-core processors:
-     `./src/manage.py test src --parallel --failfast`
-   * Only run tests from a single app, for example: `./src/manage.py test src/announcements`
-   * Only run tests from a single test class: `./src/manage.py test src.announcements.tests.test_views.AnnouncementViewTests`
-   * Only run a single test: `./src/manage.py test src.announcements.tests.test_views.AnnouncementViewTests.test_teachers_have_archive_button`
-
+     `python src/manage.py test src --parallel --failfast`
+   * Only run tests from a single app, for example: `python src/manage.py test src/announcements`
+   * Only run tests from a single test class: `python src/manage.py test src.announcements.tests.test_views.AnnouncementViewTests`
+   * Only run a single test: `python src/manage.py test src.announcements.tests.test_views.AnnouncementViewTests.test_teachers_have_archive_button`
+1. This project is configured to use git pre-commit hooks (setup using the Python "[pre-commit](https://pre-commit.com/)" module). These are run whenever a new commit is made, ensuring that all tests pass, the code is formatted properly, etc. We also have hooks that auto-correct certain simple issues with the code. It can also be useful to run these hooks manually without committing, ensuring that you have everything setup correctly:
```

Comment:
> "all tests pass" might be misleading, because precommit doesn't run the test suite, it just does a bunch of ... other tests. How to make that clear? Maybe say "ensuring the code is formatted correctly, among other checks defined in [filename here]" also add a note that some of the pre-commit hooks will auto-correct, but the commit won't be completed, so you just need to run the commit command again

My mistake. I thought that it did run the tests. I was actually confused as to why the hooks ran so quickly. I'll add a better explanation of that.